### PR TITLE
wasmdump: Dump contents of `name` section

### PR DESCRIPTION
**Pull request by sbc100**
Also add `--dump-verbose` flag to `run-wasmdump.py` in order to
be able test the output of `wasmdump -v`

### CODE REVIEW

File: src/binary-reader-objdump.c [hunk 1]
```diff
@@ -486,6 +486,21 @@ static WasmResult on_init_expr_i64_const_expr(uint32_t index,
...
+                                   void* user_data) {
+  print_details(user_data, " - [%d] " PRIstringslice "\n", index, WASM_PRINTF_STRING_SLICE_ARG(name));
+  return WASM_OK;
+}
+
+static WasmResult on_local_name(uint32_t func_index,
+                                uint32_t local_index,
+                                WasmStringSlice name,
+                                void* user_data) {
+  print_details(user_data, "  - [%d] " PRIstringslice "\n", local_index, WASM_PRINTF_STRING_SLICE_ARG(name));
```
Comment (binji) [inline src/binary-reader-objdump.c]: nit: clang-format

File: test/dump/debug-names.txt [hunk 2]
```diff
@@ -87,6 +87,29 @@
 000002a: 36                                        ; FIXUP section size
 debug-names.wasm:	file format wasm 0x00000d
 
+Section Details:
+TYPE:
+ - [0] (i32) -> nil
```
Comment (binji) [inline test/dump/debug-names.txt]: Didn't notice "nil" here, pretty weird. Probably should be `-> ()` or nothing.

File: test/dump/debug-names.txt [hunk 3]
```diff
@@ -87,6 +87,29 @@
...
+ - [1] (f32) -> nil
+FUNCTION:
+ - [0] sig=0
+ - [1] sig=1
+CODE:
+ - func 0
+ - func 1
+USER:
+ - name: "name"
+ - [0] $F1
```
Comment (binji) [inline test/dump/debug-names.txt]: It's not immediately clear what these indexes are. Maybe add "func" and "local" somewhere?

File: src/binary-reader-objdump.c (hunk 1 above)
Comment (sbc100) [inline src/binary-reader-objdump.c]: Done

File: test/dump/debug-names.txt (hunk 3 above)
Comment (sbc100) [inline test/dump/debug-names.txt]: Done

File: test/dump/debug-names.txt (hunk 2 above)
Comment (sbc100) [inline test/dump/debug-names.txt]: Hmm.. maybe a separate change

**Pull request merged**